backend/src/infra/customer-repo.ts

```diff
@@ -0,0 +1,42 @@
+import { err, ok, ResultAsync } from "neverthrow";
+import { DBCustomer } from "../domain/customer";
+import { eq } from "drizzle-orm";
+import { customers } from "../db/schema/customers";
+import { DBorTx } from "../db/db";
+import { DBInternalError } from "./shared/db-error";
+import {
+  CustomerAlreadyExistsError,
+  CustomerNotFoundError,
+} from "./customer-repo.error";
+
+export type CreatetDBCustomer = (
+  db: DBorTx
+) => (
+  id: string
```

Comment:
must: ここは DBCustomerForCreate の方がいいですね！
```suggestion
  customer: DBCustomerForCreate
```

Reply:
fixed: 5748f5cc7f271b534642a6a9ac8a3c7bce7dd906